downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md

```diff
@@ -0,0 +1,354 @@
+public function main() {
+    Identifier nonIsolatedIdentifier = new;
+
+    Identifier isolatedIdentifier = isolated object {
+        final int id = 1;
+        private string name = "default";
+
+        function updateName(string name) {
+            lock {
+                self.name = name;
+            }
+        }
+    };
+
+    boolean b1 = <any> nonIsolatedIdentifier is isolated object {}; // false
+    boolean b2 = <any> isolatedIdentifier is isolated object {}; // true
+}
+```
+
+An object-constructor expression also constructs an `isolated` object if all of the fields of the object constructor expression are `final` and the types of the fields are subtypes of `readonly` or `isolated object {}`.
+
+```ballerina
+isolated class Identifier {
+    final int i;
+    private int j;
+
+    isolated function init(int i, int j) {
+        self.i = i;
+        self.j = j;
+    }
+}
+
+public function main() {
+    var ob = object { // Constructs an `isolated` object.
+        final int a = 1000;
+        final Identifier b = new (1, 2);
+    };
+
+    isolated object {} isolatedOb = ob; // Now valid.
+}
+```
+
+##### Support for Distinct Objects
+
+###### Distinct Object Type
+
+Marking an object type definition as a `distinct` object type is supported now.
+
+```ballerina
+type Circle distinct object {
+   int x;
+   int y;
+   int r;
+};
+
+type Color [int, int, int];
+
+class ColoredCircle {
+   *Circle;
+   Color color;
+
+   function init(int x, int y, int r, Color color) {
+       self.x = x;
+       self.y = y;
+       self.r = r;
+       self.color = color;
+   }
+}
+
+class ColoredCircleLookAlike {
+   int x;
+   int y;
+   int r;
+   Color color;
+
+   function init(int x, int y, int r, Color color) {
+       self.x = x;
+       self.y = y;
+       self.r = r;
+       self.color = color;
+   }
+}
+
+Circle c = new ColoredCircle(0, 0, 0, [123, 100, 100]);
+Circle d = new ColoredCircleLookAlike(0, 0, 0, [123, 100, 100]); // Not allowed.
+```
+
+The `ColoredCircle` is a subtype of the `Circle` distinct object type even though `ColoredCircle` is not marked as a distinct class.
+
+Although `ColoredCircleLookAlike` is structurally a subtype of the `Circle` type, since `Circle` is a distinct type and `ColoredCircleLookAlike` does not have a reference to it,`ColoredCircleLookAlike` is not considered as a subtype of `Circle`.
+
+###### Distinct Object Constructor
+
+If no type reference is provided for the object constructor, contextually, the expected type must be definite and type-ids of the constructed object are the type-ids of the contextually-expected type. This allows using the object constructor to create distinct objects.
+
+#### Runtime
+
+##### Support for DynamicListener
+
+ A dynamic listener can be registered and deregistered at the runtime as follows.
+
+ ```ballerina
+ ModListener listenerVar = new(); // Where ModListener is a DynamicListener 
+	runtime:registerListener(listenerVar);
+	runtime:deregisterListener(listenerVar);
```

Comment:
Incorrect indentation for the 2nd and 3rd lines

Reply:
Fixed.